src/gpu/intel/pool/xe_global.cl

```diff
@@ -38,7 +38,7 @@ __kernel void xe_global_pooling_fwd(
     float dst_val = src[SRC_OFF(mb, oc, 0, 0, 0)];
 #endif
 #if IS_TRAINING
-    off_t max_idx = -1;
+    off_t max_idx = 0;
```

Comment:
Is the issue that `off_t(-1)` overflows the `int *ws` when storing (if no max is found)? I'm not sure what the -1 -> 0 transformation fixes

Reply:
It fixes the discrepancy between the initial assumption that the offset of the max element is `[0, 0, 0]` (see lines 35 and 37) and `max_idx` not being 0 at the start.

Since `max_idx` is only updated when some other value is greater than the one we initially assumed, there is a possibility that after sifting through the whole spatial `max_idx` remains –1, in case the initial guess is correct and `[0, 0, 0]` is indeed the global maximum.

Reply:
Here's a [rigorous proof](https://math.stackexchange.com/a/548827) of the fact that a nonempty finite set always contains elements that are its min and max.